pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,403 @@
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(1),
+					},
+				},
+			 ),
+
+		Entry("extend lease if same holder and expired lease",
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          nil,
+						RenewTime:            getTimeLease(int64(NowTime-LeaseDurationInSeconds-1)),
+						LeaseTransitions:     nil,
+						},
+				},
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{ {
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(1),
+					},
+				},
+			 ),
+		Entry("extend lease if same holder and lease will expire within two times the drainertimeout",
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          nil,
+						RenewTime:            getTimeLease(int64(NowTime) - LeaseDurationInSeconds + 2 * drainerTimeoutInMinutes * 60 - 1 ),
```

Comment:
To match the test description, you'd need to change the following:
```suggestion
						RenewTime:            getTimeLease(int64(NowTime) + drainerTimeoutInMinutes * 60),
```

`LeaseDurationInSeconds` isn't relevant here

Reply:
I updated the test title; LeaseDurationInSecond is relevent as this is setting the renew time because of the following: Expiration time := (*lease.Spec.RenewTime).Time.After( time.Duration( *lease.Spec.LeaseDurationInSeconds * time.Second) )

Reply:
The original title was fine, it was the way `RenewTime` was being calculated that seemed unnecessarily complex.

However, I see now that I was confused about what `RenewTime` represented.

`getTimeLease(int64(NowTime) - LeaseDurationInSeconds + drainerTimeoutInMinutes` would arguably be simpler but what you have is also correct.